scanpipe/pipes/d2d.py

```diff
@@ -411,21 +411,17 @@ def map_path(project, logger=None):
         _map_path_resource(to_resource, from_resources, from_resources_index)
 
 
-def create_package_from_purldb_data(project, resources, package_data):
+def get_project_resources_qs(project, resources):
     """
-    Create a DiscoveredPackage instance from PurlDB ``package_data``.
+    Return a queryset of CodebaseResources from `project` containing the
+    CodebaseResources from `resources` . If a CodebaseResource in `resources` is
+    an archive or directory, then their descendants are also included in the
+    queryset.
 
-    Return a tuple, containing the created DiscoveredPackage and the number of
-    CodebaseResources matched to PurlDB that are part of that DiscoveredPackage.
+    Return None if `resources` is empty or None.
     """
-    package_data = package_data.copy()
-    # Do not re-use uuid from PurlDB as DiscoveredPackage.uuid is unique and a
-    # PurlDB match can be found in different projects.
-    package_data.pop("uuid", None)
-    package_data.pop("dependencies", None)
-
     lookups = Q()
-    for resource in resources:
+    for resource in resources or []:
```

Comment:
Why do we need the `or []`?

Reply:
@tdruez I thought it would make the function a bit more robust in the case `resources` is None, though we don't run into that situation with the current way we use the function.